js/framework/AnalyticsConstants.js

```diff
@@ -885,6 +959,190 @@ if (!OO.Analytics.EVENT_DATA)
+   *   midrolls that play before content due to an initial playhead time > 0
+   * @property {string} plugin The video plugin used for playback
+   * @property {string} technology The browser technology used - HTML5, Flash, Mixed, or Other
+   * @property {string} encoding The stream encoding type, i.e. MP4, HLS, Dash, etc.
+   * @property {string} streamUrl The URL of the content being played
+   * @property {string} drm The DRM being used, none if there is no DRM
+   * @property {boolean} isLive Boolean that is true if a live stream is playing. If false it is VOD.
+   */
+  EVENT_DATA.InitialPlayStartingData = function(playerCoreVersion, timeSinceInitialPlay, 
+        autoplayed, hadPreroll, position, plugin, technology, encoding, streamUrl, drm, isLive)
+  {
+    var checkInitialPlayStartingData = OO._.bind(checkDataType, this, "VideoPlayerCreatedData");
+    this.playerCoreVersion = checkInitialPlayStartingData(playerCoreVersion, "playerCoreVersion", ["string"]);
+    this.timeSinceInitialPlay = checkInitialPlayStartingData(timeSinceInitialPlay, "timeSinceInitialPlay", ["number"]);
+    this.autoplayed = checkInitialPlayStartingData(autoplayed, "autoplayed", ["boolean"]);
+    this.hadPreroll = checkInitialPlayStartingData(hadPreroll, "hadPreroll", ["boolean"]);
+    this.position = checkInitialPlayStartingData(position, "position", ["number"]);
+    this.plugin = checkInitialPlayStartingData(plugin, "plugin", ["string"]);
+    this.technology = checkInitialPlayStartingData(technology, "technology", ["string"]);
+    this.encoding = checkInitialPlayStartingData(encoding, "encoding", ["object"]);
+    this.streamUrl = checkInitialPlayStartingData(streamUrl, "streamUrl", ["string"]);
+    this.drm = checkInitialPlayStartingData(drm, "drm", ["string"]);
+    this.isLive = checkInitialPlayStartingData(isLive, "isLive", ["boolean"]);
+  };
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#PlaybackReadyData
+   * @classdesc Contains the information about the playback ready event
+   * @property {string} playerCoreVersion The player core version
+   * @property {number} timeSincePlayerCreated The time between player creation and playback ready state
+   * @property {array} pluginList List of plugins loaded
+   */
+  EVENT_DATA.PlaybackReadyData = function(playerCoreVersion, timeSincePlayerCreated, pluginList)
+  {
+    var checkPlaybackReadyData = OO._.bind(checkDataType, this, "PlaybackReadyData");
+    this.playerCoreVersion = checkPlaybackReadyData(playerCoreVersion, "playerCoreVersion", ["string"]);
+    this.timeSincePlayerCreated = checkPlaybackReadyData(timeSincePlayerCreated, "timeSincePlayerCreated", ["number"]);
+    this.pluginList = checkPlaybackReadyData(pluginList, "pluginList", ["array"]);
+  };
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#ApiErrorData
+   * @classdesc Contains information about the api error.
+   * @property {object} errorCode The error code if any
```

Comment:
Should this be an object or number? Code below checks if this is a number:

this.errorCode = checkApiErrorData(errorCode, "errorCode", ["number"]);

Reply:
Should be a number, fixed